README.md

```diff
@@ -58,9 +58,13 @@ See section [recursion](/recursion) for description.
 
 *(Coming soon)*
 
-## Closures
+## [Closures](/closures)
 
-*(Coming soon)*
+Closures are functions that scope exists within a variable. Any function defined within a closure has access to the scope and variables created
+in the parent function. 
+
+Closures are a part of higher-order functions in languages such as Haskell, as well they are used in languages like JavaScript
+for callbacks, allowing the callback to access the variables and scope of the parent function.
```

Comment:
Closures are the functional equivalent to objects, they let a function carry stuff around with them in a private way.